packages/core2/src/managers/_ProjectManager.ts

```diff
@@ -52,23 +85,36 @@ export class ProjectManager extends BaseManager {
 		}
 	}
 
+	// TODO: This is a temporary strategy for upading Slice Machine configuration based on string search & replace, e.g. `this.updateSliceMachineConfig({ "200629-sms-hoy": /__PRISMIC_REPOSITORY_NAME/g })` replaces all occurence of "__PRISMIC_REPOSITORY_NAME" with "200629-sms-hoy". We'll need something stronger/more convenient for starters.
+	async updateSliceMachineConfig(
+		searchAndReplaceMap: Record<string, string | RegExp>,
+	): Promise<void> {
+		const configFilePath = await this.getSliceMachineConfigPath();
+
+		let rawConfig = await fs.readFile(configFilePath, "utf-8");
+
+		for (const replacement in searchAndReplaceMap) {
+			const searchPattern = searchAndReplaceMap[replacement];
+
+			rawConfig = rawConfig.replace(searchPattern, replacement);
+		}
+
+		await fs.writeFile(configFilePath, rawConfig);
+		delete this._cachedSliceMachineConfig; // Clear config cache
+	}
+
 	async loadSliceMachineConfig(): Promise<SliceMachineConfig> {
 		// TODO: Reload plugins with a fresh plugin runner. Plugins may
 		// have been added or removed.
 
-		const projectRoot = await this.getRoot();
+		const configFilePath = await this.getSliceMachineConfigPath();
 
 		let configModule: unknown | undefined;
-
-		for (const configFileName of SLICE_MACHINE_CONFIG_FILENAMES) {
-			const configFilePath = path.resolve(projectRoot, configFileName);
-
-			try {
-				await fs.access(configFilePath);
-				configModule = loadModuleWithJiti(path.resolve(configFileName));
-			} catch {
-				// noop
-			}
+		try {
+			await fs.access(configFilePath);
+			configModule = loadModuleWithJiti(path.resolve(configFilePath));
```

Comment:
Per Côme, we'll be sticking with `sm.json` for now. We'll need to refactor this file to read the JSON file instead of `slicemachine.config.ts`.

This means we can update the repository name with property replacement rather than search/replace.

Reply:
Could we at least support a `slicemachine.config.json` for new projects? `sm.json` feels suboptimal for a name :/

Reply:
I like that. It's clearer than `sm.json` and sets us up for supporting the `.js`/`.ts` version.